src/rgw/rgw_admin.cc

```diff
@@ -1585,13 +1585,20 @@ static boost::optional<RGWRESTConn> get_remote_conn(RGWRados *store,
   return conn;
 }
 
-#define MAX_REST_RESPONSE (128 * 1024) // we expect a very small response
+// we expect a very small response
+static constexpr size_t MAX_REST_RESPONSE = 128 * 1024;
+
 static int send_to_remote_gateway(RGWRESTConn* conn, req_info& info,
                                   bufferlist& in_data, JSONParser& parser)
 {
-  bufferlist response;
+  if (!conn) {
+    return -EINVAL;
+  }
```

Comment:
This checking is fine for this function. But it looks like the test doesn't identify the branching correctly. I have sent you an email about it.

`http://people.redhat.com/bhubbard/scan-build-2017-06-14-192846-5371-1/report-5828b7.html#Path14`